lms/djangoapps/certificates/views.py

```diff
@@ -224,3 +231,76 @@ def update_example_certificate(request):
 
     # Let the XQueue know that we handled the response
     return JsonResponse({'return_code': 0})
+
+
+@login_required
+def render_html_view(request):
+    """
+    This view generates an HTML representation of the specified student's certificate
+    If a certificate is not available, we display a "Sorry!" screen instead
+    """
+    invalid_template_path = 'certificates/invalid.html'
+
+    # Feature Flag check
+    if not settings.FEATURES.get('CERTIFICATES_HTML_VIEW', False):
+        return render_to_response(invalid_template_path)
+
+    context = {}
+    course_id = request.GET.get('course', None)
+    context['course'] = course_id
+    if not course_id:
+        return render_to_response(invalid_template_path, context)
+
+    # Course Lookup
+    try:
+        course_key = CourseKey.from_string(course_id)
+    except InvalidKeyError:
+        return render_to_response(invalid_template_path, context)
+    course = modulestore().get_course(course_key)
+    if not course:
+        return render_to_response(invalid_template_path, context)
+
+    # Certificate Lookup
+    try:
+        certificate = GeneratedCertificate.objects.get(
+            user=request.user,
+            course_id=course_key
+        )
+    except GeneratedCertificate.DoesNotExist:
+        return render_to_response(invalid_template_path, context)
+
+    # Load static output values from configuration,
+    configuration = CertificateHtmlViewConfiguration.get_config()
+    context = configuration.get('default', {})
+    # Override the defaults with any mode-specific static values
+    context.update(configuration.get(certificate.mode, {}))
+    # Override further with any course-specific static values
+    context.update(course.cert_html_view_overrides)
+
+    # Populate dynamic output values using the course/certificate data loaded above
+    user_fullname = request.user.profile.name
+    company_name = context.get('company_name')
+    context['accomplishment_copy_name'] = user_fullname
+    accd = 'a course of study offered by <span class="detail--xuniversity">{0}</span>'.format(course.org)
+    accd = accd + ', through <span class="detail--company">{0}</span>.'.format(company_name)
```

Comment:
please don't concat strings for internationalization. or use html. please do something like,
```
# Translators: This phrase appears after a heading that consists of a course name, for example: \n
# Potions\n
# a course of study offered by Hogwarts, through edX
'a course of study offered by {university_name}, through {platform_name}'.format(university_name=u'<span class...'...)
```

The "\n" are meant to be literal in that example - translators comments appear all on one line on the Transifex site unless you add in the \n explicitly.